vllm/v1/worker/gpu_model_runner.py

```
@@ -304,6 +257,31 @@ def __init__(
                                         pin_memory=self.pin_memory)
         self.seq_lens_np = self.seq_lens_cpu.numpy()
 
+    def _may_reorder_batch(self, scheduler_output: "SchedulerOutput") -> bool:
+        """
+        Update the order of requests in the batch based on the attention
+        backend's needs. For example, some attention backends (namely MLA) may 
+        want to separate requests based on if the attention computation will be
+        compute-bound or memory-bound.
+
+        Args:
+            scheduler_output: The scheduler output.
+
+        Returns:
+            True if the batch was reordered, False otherwise.
+        """
+        batch_reordered = self.attn_metadata_builders[0].reorder_batch(
```

should we skip below if the first group does not need to re-order to reduce overhead?

---

For example, the first group is flashattn, and the second group is flashmla. We can't handle this case and should raise an error. To accelerate, I plan to introduce something like the following in the future, to know the expect order of each metadata builder 
```
class BatchOrder(enum.Enum):
    any_order, prefill_first, decode_first
```